analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Pattern.cs

```diff
@@ -34,25 +35,32 @@ internal static class Pattern
             ? context.SetOperationConstraint(newConstraint)
             : context.State;
 
-        public static ProgramState Process(SymbolicContext context, IRecursivePatternOperationWrapper recursive)
+        public static ProgramState Process(SymbolicContext context, IRecursivePatternOperationWrapper recursive) =>
+            ProcessDeclaration(context, recursive.DeclaredSymbol, true);
+
+        public static ProgramState Process(SymbolicContext context, IDeclarationPatternOperationWrapper declaration) =>
+            ProcessDeclaration(context, declaration.DeclaredSymbol, !declaration.MatchesNull);  // "... is var ..." should not set NotNull
```

Comment:
`MatchesNull` nice solution! NIT: You may name the parameter `matchesNull` instead of `setNotNull` to avoid the negation and be consistent with the Roslyn API.

Reply:
`matchesNull` doesn't make sense for the `RecursivePattern` scenario. So I named it based on what it does.